File: 0169-majority-element/solution.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int el = 0,fre = 0;
        for (int i = 0; i<nums.size(); i++){
            fre = 0;
            for (int  j = 0; j<nums.size(); j++){
                if(nums[i] == nums[j]){
                    fre++;
                }
                if (fre>nums.size()/2){
                    return nums[i];
                }
            }
        }
        return -1;
    }
};

File: 0169-majority-element/solution.java
class Solution {
    public int majorityElement(int[] nums) {
        int curr = 0;
        int cnt = 0;

        for (int i = 0; i < nums.length; i++){
            if(cnt == 0){
                curr = nums[i];
                cnt++;
            }else if(nums[i] == curr)cnt++;
            else cnt--;
        }

        int cnt1 = 0;
        for(int i = 0; i < nums.length; i++){
            if(nums[i] == curr)cnt1++;
            if(cnt1 > (nums.length/2))return curr;
        }
        return -1;
    }
}
